### PR TITLE
added success count and lead time to an overall account level metric

### PR DESCRIPTION
**Pull request by zmaleki**

### CODE REVIEW

File: src/pipelineEventHandler.js [hunk 1]
```diff
@@ -8,183 +8,202 @@ const COUNT = 'Count';
...
+  }
 
-    putMetricData(state) {
-        if (state.metricData.length > 0) {
-            return state.cloudwatch.putMetricData({
-                Namespace: 'Pipeline',
-                MetricData: state.metricData
-            }).promise();
-        }
+  static addMetric(state, metricName, unit, value, addAccountMetric = false) {
```
Comment (temyers) [inline src/pipelineEventHandler.js]: The parameter means this method is now doing two things:
* refactor into separate methods so each 'does one thing'
* A separate parameter could be 'confusing for the client' - why would I choose true/false.

```
PipelineEventHandler.addMetric(state, 'DeliveryLeadTime', SECONDS, leadDuration);
PipelineEventHandler.addAccountMetric(state, 'DeliveryLeadTime', SECONDS, leadDuration);
```

Comment (temyers) [reply src/pipelineEventHandler.js]: Understand why you did it - so it's a single SDK call.

Instead, suggest:

`createMetricData` for each (?) better name....

Something like..
```
postMetrics(metricData(...),accountMetricData(...))
```

If there are shared params, etc, then turning the params into an object probably makes sense?

File: test/index.test.js [hunk 2]
```diff
@@ -2,835 +2,867 @@
...
+    },
+  ].forEach(scenario => {
+    if (scenario.skip) {
+      return;
+    }
+    it(`handles scenario successfully - ${scenario.description}`, () => {
+      let putMetricDataSpy = sinon.spy();
+
+      AWS.mock('CloudWatch', 'putMetricData', (params, cb) => {
+        console.log('Params are: ', JSON.stringify(params));
```
Comment (temyers) [inline test/index.test.js]: console?

File: test/index.test.js [hunk 3]
```diff
@@ -2,835 +2,867 @@
...
-                        });
-                })
-            }
-        })
-    })
-
-})
-
+              });
+              console.log(pipelineNames);
```
Comment (temyers) [inline test/index.test.js]: remove

**Review by temyers — CHANGES_REQUESTED** [review]

File: test/index.test.js (hunk 2 above)
Comment (zmaleki) [reply test/index.test.js]: done

File: test/index.test.js (hunk 3 above)
Comment (zmaleki) [reply test/index.test.js]: done

**Pull request merged**